src/critics/reference_trajectory_critic.cpp

```diff
@@ -92,13 +99,65 @@ xt::xtensor<double,
           P[0] = P1(s, 0) + u * P2_P1_diff(s, 0);
           P[1] = P1(s, 1) + u * P2_P1_diff(s, 1);
         }
-        min_dist = std::min(evaluate_dist(std::move(P), t, p), min_dist);
+        auto dist = evaluate_dist(std::move(P), t, p);
+        if (dist < min_dist) {
+          min_s = s;
+          min_dist = dist;
+        }
       }
+      max_s = std::max(max_s, min_s);
       mean_dist += min_dist;
     }
-    distances(t) = mean_dist / trajectories_points_count;
+
+    cost(t) = mean_dist / trajectories_points_count;
+  }
+
+  costs += xt::pow(cost * reference_cost_weight_, reference_cost_power_);
+
+  if (enable_nearest_path_angle_critic_) {
+    cost = xt::zeros<double>({trajectories_count});
+
```

Comment:
Are these 2 exactly the same as the path angle and the goal critics? If so, why not actually have instances of those objects as members in this class and then call them here, eg

`cost += path_angle_critic_.score(args) + goal_critic_.score(args)`

Reply:
They are not the same. see above